coordinator/gscoordinator/utils.py

```diff
@@ -1211,7 +1286,77 @@ def _extract_gar(app_dir: str, attr):
         zip_ref.extractall(app_dir)
 
 
-def _codegen_app_info(attr, meta_file: str):
+def _parse_giraph_app_type(java_class_path, real_algo):
+    _java_app_type = ""
+    _frag_param_str = ""
+    _java_inner_context_type = ""
+    _java_executable = "java"
+    if shutil.which("java") is None:
```

Comment:
Already has utils function to check java exist.

https://github.com/alibaba/GraphScope/blob/main/python/graphscope/framework/utils.py#L189

Reply:
Ok